### PR TITLE
Use decoder package from PyPI (Fixes #3044)

### LINKED ISSUE #3044: Publish decoder package on PyPI to avoid pip edge cases
Our current setup with a direct URL dependency on the decoder package requires the use of `--force-reinstall` when installing our training code, which reinstalls *everything*, not just upgraded packages. This is a pretty awkward setup, but required because pip has completely broken version handling with direct URL dependencies.

Two ways we make this better are:

  - Publishing the decoder package on PyPI
  - Publishing a simple HTML file with links to the various wheels and using `pip install --find-links`

Because I'm lazy, and the first option requires less work, I vote for it. All we have to do is publish the packages and then use a normal dependency specification. `pip install --upgrade` (and even a plain `pip install -e`, I think) should then pick up upgrades without reinstalling the world.

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,20 +25,15 @@ def get_decoder_pkg_url(version, artifacts_root=None):
...
+        m_or_mu = 'mu' if is_ucs2 else 'm'
 
     pyver = ''.join(str(i) for i in sys.version_info[0:2])
 
-    if not artifacts_root:
-        artifacts_root = 'https://community-tc.services.mozilla.com/api/index/v1/task/project.deepspeech.deepspeech.native_client.{branch_name}.{tc_arch_string}/artifacts/public'.format(
-            branch_name=branch,
-            tc_arch_string=tc_arch)
-
     return 'ds_ctcdecoder @ {artifacts_root}/ds_ctcdecoder-{ds_version}-cp{pyver}-cp{pyver}{m_or_mu}-{platform}_{arch}.whl'.format(
```
Comment (lissyx) [inline setup.py:37]: do we want to keep the name like that?

File: taskcluster/tc-train-tests.sh [hunk 2]
```diff
@@ -8,6 +8,13 @@ extract_python_versions "$1" "pyver" "pyver_pkg" "py_unicode_type" "pyconf" "pya
 
 bitrate=$2
 
+decoder_src=$3
+
+if [ "$decoder_src" = "--pypi" ]; then
```
Comment (lissyx) [inline taskcluster/tc-train-tests.sh:13]: nit: `--pypi` is confusing, I was expecting the code to indeed fetch from pypi, can we change to `--no-pypi` ?

File: setup.py (hunk 1 above)
Comment (reuben) [reply setup.py:37]: like what?

File: taskcluster/tc-train-tests.sh (hunk 2 above)
Comment (reuben) [reply taskcluster/tc-train-tests.sh:13]: That's what it does, fetch from PyPI. I guess the code is even more confusing then. What's leading you to believe that this flag is "disabling PyPI"?

File: setup.py (hunk 1 above)
Comment (lissyx) [reply setup.py:37]: something that does not include "ds" maybe :)

File: taskcluster/tc-train-tests.sh (hunk 2 above)
Comment (lissyx) [reply taskcluster/tc-train-tests.sh:13]: Commenting after waking up, I guess ...

**Pull request merged**